### PR TITLE
Add macOS DMGs to build.yml

### PR DESCRIPTION
**Pull request by TheRealJayjayli**

### CODE REVIEW

**Comment by TheRealJayjayli** [comment]
For some reason, macOS still thinks the file produced is damaged. I think it's because Apple requires the application to be signed. I can try taking a look into this later.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -96,3 +96,40 @@ jobs:
         with:
           name: linux-zip
           path: ./chunkyImage/**/*
+
+  macOS:
```
Comment (leMaik) [inline .github/workflows/build.yml]:
```suggestion
  macos:
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -96,3 +96,40 @@ jobs:
...
+      - name: Build DMG package
+        run: >
+          jpackage --type dmg --runtime-image chunkyjava --input jars/
+          --main-jar ${{ steps.launcherjar.outputs.filename }} --name Chunky
+          --vendor leMaik --app-version 1.12.1 --copyright "Copyright (c) 2010-2021 Chunky contributors"
+          --icon chunky.icns
+          --dest chunkyImage
+      - uses: actions/upload-artifact@v2
+        with:
+          name: macOS-DMG
```
Comment (leMaik) [inline .github/workflows/build.yml]:
```suggestion
          name: macos-dmg
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -96,3 +96,40 @@ jobs:
...
+        with:
+          url: https://chunkyupdate.lemaik.de/ChunkyLauncher.jar
+          target: jars
+      - name: Build DMG package
+        run: >
+          jpackage --type dmg --runtime-image chunkyjava --input jars/
+          --main-jar ${{ steps.launcherjar.outputs.filename }} --name Chunky
+          --vendor leMaik --app-version 1.12.1 --copyright "Copyright (c) 2010-2021 Chunky contributors"
+          --icon chunky.icns
+          --dest chunkyImage
```
Comment (leMaik) [inline .github/workflows/build.yml]: Would this still work?

```suggestion
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -96,3 +96,40 @@ jobs:
...
+      - uses: actions/upload-artifact@v2
+        with:
+          name: macOS-DMG
+          path: ./chunkyImage/*.dmg
```
Comment (leMaik) [inline .github/workflows/build.yml]: ...when we also change the path here?

```suggestion
          path: ./*.dmg
```

**Review by leMaik — APPROVED** [review]

File: .github/workflows/build.yml (hunk 4 above)
Comment (TheRealJayjayli) [reply .github/workflows/build.yml]: It should still work 👍

**Comment by TheRealJayjayli** [comment]
Oh... I need to actually press commit changes? My bad!

**Pull request merged**